newton/_src/solvers/vbd/rigid_vbd_kernels.py

```
@@ -2730,6 +2871,69 @@ def apply_angular_drive_limit_torque(
+        return wp.vec3(0.0), wp.mat33(0.0)
+
+    kappa = kappa_cached
+    J_world = J_world_cached
+    if not has_cached:
+        kappa, J_world = compute_kappa_and_jacobian(q_wp, q_wc, q_wp_rest, q_wc_rest)
+
+    theta = wp.dot(kappa, axis)
+    theta_abs = theta + rest_angle
+    omega_p = quat_velocity(q_wp, q_wp_prev, dt)
+    omega_c = quat_velocity(q_wc, q_wc_prev, dt)
+    dkappa_dt = compute_kappa_dot_analytic(q_wp, q_wc, q_wp_rest, q_wc_rest, omega_p, omega_c, kappa)
```

Ruff (0.16.3)

[error] Undefined name `compute_kappa_dot_analytic`

(F821)